awx/main/models/ha.py

```diff
@@ -261,18 +261,20 @@ class Meta:
         app_label = 'main'
 
 
-    def fit_task_to_most_remaining_capacity_instance(self, task):
+    @staticmethod
+    def fit_task_to_most_remaining_capacity_instance(task, instances):
         instance_most_capacity = None
-        for i in self.instances.filter(capacity__gt=0, enabled=True).order_by('hostname'):
+        for i in instances:
```

Comment:
@chrismeyersfsu @fosterseth to address my race condition comment below, perhaps we should sprinkle in some sort of `refresh_from_db` call with an exception block to make _absolutely sure_ the selected instance *actually* still exists?